### DISCUSSION TITLE
Use conformer in AISHELL1 recipes

### DISCUSSION THREAD

**Discussion by toby1123yjh** [speechbrain/speechbrain#1689] [Q&A] [answered]
Based on the recipes of Librispeech comformer.yaml, I add the following code to the Transformer settings of the transformer.yaml of aishell1 recipes to use conformer encoder. 
But the model converges very slowly on 2 RTX2080Ti: the valid CER only reach 10.05 when I trained the model for 50 epochs.
Due to the lack of computing resources, I don't know whether my code is correct if I want to use conformer encoder, 
or whether I should modify other parameters?

the hparam file is based on the recipe of aishell1 train_ASR_transformer.yaml. 
In order to adapt to my server hardware, I only modified a few hyparameters:
1. batch_size: 8 -> 4(now)
2. gradient_accumulation: 4 -> 8(now)
3. encoder_module: conformer(add)
4. attention_type: RelPosMHAXL(add)
5. causal: False(add)

**Comment by BenoitWang** [comment]
Hi @toby1123yjh did you solve it? What are the hparams that you modified? And maybe take a look at the optimiser and learning rate as well.

**Reply by toby1123yjh** [reply]
the hparam file is based on the recipe of aishell1 train_ASR_transformer.yaml.
In order to adapt to my server hardware, I only modified a few hyparameters:

batch_size: 8 -> 4(now)
gradient_accumulation: 4 -> 8(now)
encoder_module: conformer(add)
attention_type: RelPosMHAXL(add)
causal: False(add)

**Reply by BenoitWang** [reply]
Hi @toby1123yjh, very sorry to make you repeat... My idea for now is to change the optimiser to see what happens. For the aishell transformer recipe the optimiser is in a two-stage setting, maybe try to modify it according to the librispeech recipe?

**Reply by toby1123yjh** [reply]
Thank you for your reply! I changed the optimizer to just use adam, and the lr_adam was set to 0.001 based on the librispeech recipe.
The Cer can only converge to 9.3.
So I changed the lr to 0.01、0.005、0.002 based on your reply. But the result is worse than 0.001, which i think the poor performance is not caused by the learning rate or optimiser. 
Do you think batchsize is the reason？
From my historical experience of using speech brain, it seems that I can't set too small batchsize for the ASR model. like less than 8.

**Reply by BenoitWang** [reply]
I think with gradient accumulation it should be ok. Some other suggestions would be:
1. increase gradient accumulation
2. try sorting the data or use dynamic batching
3. train a smaller conformer to see how it converges

**Reply by toby1123yjh** [reply]
I will carry out some experiments based on these opinions as soon as possible. Thank you for your suggestions and all your replies!!. :relaxed:

**Comment by toby1123yjh** [comment, marked as answer]
After several days of validation experiments in my experimental environment(2 RTX 2080 TI，RAM 11G), I found that using DDP on the speechbrain may affect the final performance of the model. At the same time, in order to obtain the expected results, the batch_size and accumulation also need to modify when using the conformer encoder in 1 RTX 2080 TI.
